elements/lisk-p2p/test/unit/peer/outbound.ts

```diff
@@ -0,0 +1,240 @@
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import * as querystring from 'querystring';
+import * as socketClusterClient from 'socketcluster-client';
+import { expect } from 'chai';
+import { OutboundPeer } from '../../../src/peer';
+import { DEFAULT_WS_MAX_PAYLOAD } from '../../../src/constants';
+import { sanitizeNodeInfoToLegacyFormat } from '../../../src/utils';
+import { REMOTE_SC_EVENT_MESSAGE } from '../../../src/events';
+import { SCClientSocket } from 'socketcluster-client';
+
+describe('peer/outbound', () => {
+	const DEFAULT_RANDOM_SECRET = 123;
+	const defaultPeerInfo = {
+		ipAddress: '12.12.12.12',
+		wsPort: 5001,
+		height: 545776,
+		isDiscoveredPeer: true,
+		version: '1.1.1',
+		protocolVersion: '1.1',
+	};
+	const defaultOutboundPeerConfig = {
+		rateCalculationInterval: 1000,
+		wsMaxMessageRate: 1000,
+		wsMaxMessageRatePenalty: 10,
+		secret: DEFAULT_RANDOM_SECRET,
+		maxPeerInfoSize: 10000,
+		maxPeerDiscoveryResponseLength: 1000,
+	};
+	const nodeInfo = {
+		os: 'os',
+		version: '1.2.0',
+		protocolVersion: '1.2',
+		nethash: 'nethash',
+		wsPort: 6001,
+		height: 100,
+	};
+	const legacyNodeInfo = nodeInfo
+		? sanitizeNodeInfoToLegacyFormat(nodeInfo)
+		: undefined;
+	const clientOptions = {
+		hostname: '12.12.12.12',
+		port: 5001,
+		query: querystring.stringify({
+			...legacyNodeInfo,
+			options: JSON.stringify(legacyNodeInfo),
+		}),
+		connectTimeout: 500,
+		ackTimeout: 500,
+		multiplex: false,
+		autoConnect: false,
+		autoReconnect: false,
+		maxPayload: DEFAULT_WS_MAX_PAYLOAD,
+	};
+	let defaultOutboundPeer: OutboundPeer;
+
+	beforeEach(() => {
+		defaultOutboundPeer = new OutboundPeer(
+			defaultPeerInfo,
+			defaultOutboundPeerConfig,
+		);
+	});
+
+	afterEach(() => {
+		sandbox.restore();
+		defaultOutboundPeer.disconnect();
+	});
+
+	describe('#constructor', () => {
+		it('should be an instance of OutboundPeer class', () =>
+			expect(defaultOutboundPeer).to.be.instanceof(OutboundPeer));
+	});
+
+	describe('#socket', () => {
+		it('should unbind handlers from outbound socket if it exists', () => {
+			const outboundSocket = socketClusterClient.create(clientOptions);
+			sandbox.stub(
+				defaultOutboundPeer as any,
+				'_unbindHandlersFromOutboundSocket',
+			);
+
+			(defaultOutboundPeer as any)._socket = outboundSocket;
+			expect((defaultOutboundPeer as any)._socket).to.eql(outboundSocket);
+			defaultOutboundPeer.socket = outboundSocket;
+			expect((defaultOutboundPeer as any)._unbindHandlersFromOutboundSocket).to
+				.be.calledOnce;
+		});
+		it('should set new socket', () => {
+			const outboundSocket = socketClusterClient.create(clientOptions);
```

Comment:
doesn't this socket hangs around if we create the real socket? same for other cases.
Maybe it's good to mock it.